lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -782,10 +782,11 @@ SPIRVFunction *LLVMToSPIRVBase::transFunctionDecl(Function *F) {
   BF->setFunctionControlMask(transFunctionControlMask(F));
   if (F->hasName()) {
     if (isKernel(F)) {
-      /* strip the prefix as the runtime will be looking for this name */
-      std::string Prefix = kSPIRVName::EntrypointPrefix;
-      std::string Name = F->getName().str();
-      BM->setName(BF, Name.substr(Prefix.size()));
+      // Strip the entry point wrapper prefix as the runtime will be looking for
+      // this name if found if there is the prefix
+      StringRef Name = F->getName();
+      (void) Name.consume_front(kSPIRVName::EntrypointPrefix);
```

Comment:
cppreference says:
While the behavior of std::ignore outside of [std::tie](https://en.cppreference.com/w/cpp/utility/tuple/tie) **is not formally specified**, some code guides recommend using std::ignore to avoid warnings from unused return values of [[[nodiscard](https://en.cppreference.com/w/cpp/language/attributes/nodiscard)]] functions.

TBH I'd prefer not to rely on 'not formally specified' behavior.

Reply:
OK, agree